### PR TITLE
[1.8] support mixed network multi cluster

### PR DESCRIPTION
**Pull request by waynz0r**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### What's in this PR?

- implements support for setting network for remote clusters.
- use proper service ports on remote clusters 
- trigger `MeshGateway` reconcile on `Istio` CR change
- support for mesh expansion gateway using local endpoints only (needs Backyards Istio distribution for the actual feature)

### CODE REVIEW

File: pkg/apis/istio/v1beta1/remoteistio_types.go [hunk 1]
```diff
@@ -80,6 +80,10 @@ type RemoteIstioSpec struct {
 	// Istio CR to which this gateway belongs to
 	IstioControlPlane *NamespacedName `json:"istioControlPlane,omitempty"`
 
+	// Network defines the network this cluster belong to. This name
```
Comment (Laci21) [inline pkg/apis/istio/v1beta1/remoteistio_types.go:83]:
```suggestion
	// Network defines the network this cluster belongs to. This name
```

File: pkg/controller/istio/istio_controller.go [hunk 2]
```diff
@@ -477,14 +480,19 @@ func (r *ReconcileIstio) getMeshNetworks(config *istiov1beta1.Istio, logger logr
...
+		// to be backward compatible, the network name is the CR name if not specified
+		if networkName == "" {
+			networkName = remoteIstio.Name
+		}
+
+		if meshNetworks[networkName] == nil {
+			meshNetworks[networkName] = &istiov1beta1.MeshNetwork{}
 		}
+		meshNetworks[networkName].Endpoints = append(meshNetworks[networkName].Endpoints, istiov1beta1.MeshNetworkEndpoint{
+			FromRegistry: remoteIstio.Name,
```
Comment (Laci21) [inline pkg/controller/istio/istio_controller.go:493]: Might be missing something here, but shouldn't this be:
```suggestion
			FromRegistry: networkName,
```

File: pkg/resources/istiod/service.go [hunk 3]
```diff
@@ -21,44 +21,49 @@ import (
...
 
+	"github.com/banzaicloud/istio-operator/pkg/apis/istio/v1beta1"
 	"github.com/banzaicloud/istio-operator/pkg/resources/templates"
 	"github.com/banzaicloud/istio-operator/pkg/util"
 )
 
+func ServicePorts(config *v1beta1.Istio) []apiv1.ServicePort {
+	return []apiv1.ServicePort{
+		{
+			// plaintext
```
Comment (Laci21) [inline pkg/resources/istiod/service.go:32]: Nit: 
```suggestion
		// plaintext
```

File: pkg/controller/istio/istio_controller.go (hunk 2 above)
Comment (waynz0r) [reply pkg/controller/istio/istio_controller.go:493]: The registry is the name of the cluster, not the network.

**Review by khernyo — APPROVED** [review]

File: pkg/controller/istio/istio_controller.go (hunk 2 above)
Comment (Laci21) [reply pkg/controller/istio/istio_controller.go:493]: got it, thanks

**Review by Laci21 — APPROVED** [review]

**Pull request merged**